firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/Serial/Nonin/FrameReceiver.cpp

```diff
@@ -46,87 +43,64 @@ bool validate_start_of_frame(const Frame &new_frame) {
   if (new_frame[0] == 0x01 && (new_frame[1] & mask_start_of_frame) == mask_start_of_frame) {
-    /* Checksum validation */
     if (((new_frame[0] + new_frame[1] + new_frame[2] + new_frame[3]) % (UINT8_MAX + 1)) ==
         new_frame[4]) {
-      /* Return the start of packet status as available */
-      return FrameReceiver::FrameInputStatus::available;
+      return FrameInputStatus::output_ready;
     }
   }
 
-  /* return the frame status as not available */
-  return FrameReceiver::FrameInputStatus::framing_error;
+  return FrameInputStatus::checksum_failed;
 }
 
-FrameReceiver::FrameInputStatus FrameReceiver::update_frame_buffer(uint8_t new_byte) {
+FrameInputStatus FrameReceiver::update_frame_buffer(uint8_t new_byte) {
   Frame frame_buffer;
 
-  /* Input the new byte received and check for frame availability */
-  if (frame_buf_.input(new_byte) == BufferStatus::partial) {
-    /* return false on frame is not available */
-    return FrameInputStatus::waiting;
+  if (frame_buf_.input(new_byte) == FrameBufferStatus::waiting) {
+    return FrameInputStatus::ok;
   }
 
-  /* On frame buffer input status is available update the frameBuffer with new frame */
-  if (frame_buf_.output(frame_buffer) != BufferStatus::ok) {
-    /* return false on frame is not available */
-    return FrameInputStatus::waiting;
+  if (frame_buf_.output(frame_buffer) != FrameBufferStatus::ok) {
+    return FrameInputStatus::ok;
   }
 
-  /* On start_of_frame_status_ false Validate the start of frame */
   if (!start_of_frame_status_) {
-    /* Validate the start of frame in the beginning of reading sensor data and
-       on there is loss of bytes in a frame or noise occurred in recived frame
-       due to which the validation of start of frame is called */
     if (validate_start_of_frame(frame_buffer)) {
-      /* On start of frame available update the start frame status as true */
       start_of_frame_status_ = true;
-      /* On Start frame is available return status as available */
-      return FrameInputStatus::available;
+      return FrameInputStatus::output_ready;
     }
-    /* On non available of start frame left shift the frame buffer */
     frame_buf_.shift_left();
-    /* On Start frame is not available return status as waiting */
-    return FrameInputStatus::waiting;
+    return FrameInputStatus::ok;
   }
 
-  /* Validate the frame received and return the status */
   input_status_ = validate_frame(frame_buffer);
-  if (input_status_ == FrameInputStatus::framing_error) {
-    /* On checksum error update the start frame status as false */
+  if (input_status_ == FrameInputStatus::checksum_failed) {
     start_of_frame_status_ = false;
   }
+
   return input_status_;
 }
 
-FrameReceiver::FrameInputStatus FrameReceiver::input(const uint8_t new_byte) {
-  /* Update the frame buffer with new byte received */
+FrameInputStatus FrameReceiver::input(const uint8_t new_byte) {
   input_status_ = this->update_frame_buffer(new_byte);
```

Comment:
[for next PR] I don't think we need to use `this->`, and in general we don't use `this->` in our codebase.